Sporadic failures of CI in test_mass_flow_controller

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

<!-- A clear and concise description of what the bug is. -->

Recent CI runs show spurious failures of `test_reactor.TestIdealGasReactor.test_mass_flow_controller` in the Sundials suite. It appears that failures do not occur for every run: thus far, they most often occurred for Sundials 5.7, but at least one failure occurred for Sundials 3.

**Steps to reproduce**

<!-- A minimal, complete, and reproducible example demonstrating the problem. See https://stackoverflow.com/help/minimal-reproducible-example for additional suggestions on how to create such an example. -->

E.g. see recent GH Actions failures in #1032

**Behavior**

<!-- Describe the result of executing the above steps, and how this differs from what you expect to happen. -->

Test fails with 
```
======================================================================
FAIL: test_mass_flow_controller (cantera.test.test_reactor.TestIdealGasReactor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cantera/cantera/build/python/cantera/test/test_reactor.py", line 478, in test_mass_flow_controller
    self.assertArrayNear(ma * Ya + 0.1 * gas2.Y, mb * Yb)
  File "/home/runner/work/cantera/cantera/build/python/cantera/test/utilities.py", line 82, in assertArrayNear
    self.fail(worst[1])
AssertionError: AssertNear: 0.1 - 0.10000000665769 = -6.6576915647154e-09
Relative error for element (0,) of 6.651040e-08 exceeds rtol = 1.000000e-08

----------------------------------------------------------------------
```

**System information**

- Cantera version: latest `main` (19f1593476112279d2d8045ba09f193038335bc1)
- OS: GH Actions

--- a/interfaces/cython/cantera/test/test_reactor.py
+++ b/interfaces/cython/cantera/test/test_reactor.py
@@ -443,7 +443,13 @@ class TestReactor(utilities.CanteraTest):
         reservoir = ct.Reservoir(gas2)
 
         mfc = ct.MassFlowController(reservoir, self.r1)
-        mfc.mass_flow_rate = lambda t: 0.1 if 0.2 <= t < 1.2 else 0.0
+        # Triangular pulse with area = 0.1
+        def mdot(t):
+            if 0.2 <= t < 1.2:
+                return 0.2 - 0.4 * abs(t - 0.7)
+            else:
+                return 0.0
+        mfc.mass_flow_rate = mdot
         self.assertEqual(mfc.mass_flow_coeff, 1.)
 
         self.assertEqual(mfc.type, type(mfc).__name__)
@@ -462,10 +468,10 @@ class TestReactor(utilities.CanteraTest):
 
         self.net.advance(0.1)
         self.assertNear(mfc.mass_flow_rate, 0.)
-        self.net.advance(0.2)
-        self.assertNear(mfc.mass_flow_rate, 0.1)
-        self.net.advance(1.1)
-        self.assertNear(mfc.mass_flow_rate, 0.1)
+        self.net.advance(0.3)
+        self.assertNear(mfc.mass_flow_rate, 0.04)
+        self.net.advance(1.0)
+        self.assertNear(mfc.mass_flow_rate, 0.08)
         self.net.advance(1.2)
         self.assertNear(mfc.mass_flow_rate, 0.)
 

--- a/interfaces/cython/cantera/test/test_reactor.py
+++ b/interfaces/cython/cantera/test/test_reactor.py
@@ -442,6 +442,13 @@ class TestReactor(utilities.CanteraTest):
         gas2.TPX = 300, 10*101325, 'H2:1.0'
         reservoir = ct.Reservoir(gas2)
 
+        # Apply a mass flow rate that is not a smooth function of time. This
+        # demonstrates the accuracy that can be achieved by having the mass flow rate
+        # evaluated simultaneously with the rest of the governing equations, as opposed
+        # to doing only between integrator time steps. In the latter case, larger
+        # errors would be introduced when the integrator steps past the times where the
+        # function's behavior changes and can't dynamically reduce the steps size for
+        # the already-completed steps.
         mfc = ct.MassFlowController(reservoir, self.r1)
         # Triangular pulse with area = 0.1
         def mdot(t):

